nvflare/private/fed/client/client_runner.py

```diff
@@ -398,11 +406,32 @@ def _do_process_task(self, task: TaskAssignment, fl_ctx: FLContext, abort_signal
         return self._reply_and_audit(reply=reply, ref=server_audit_event_id, fl_ctx=fl_ctx, msg="submit result OK")
 
     def _try_run(self):
+        heartbeat_thread = threading.Thread(target=self._send_job_heartbeat, args=[], daemon=True)
+        heartbeat_thread.start()
+
         while not self.run_abort_signal.triggered:
             with self.engine.new_context() as fl_ctx:
                 task_fetch_interval, _ = self.fetch_and_run_one_task(fl_ctx)
             time.sleep(task_fetch_interval)
 
+    def _send_job_heartbeat(self, interval=30.0):
+        wait_times = int(interval / 2)
```

Comment:
maybe we can make this 2 becomes a variable? it corresponds to the sleeping time right?
`poll_time=2`?

Reply:
This was how it was implemented in 2.3. Basically the idea here is that you want to send a HB every "interval" seconds, but you don't want to wait for that long between HBs (since this will block the thread from doing anything while sleeping). We want to be able to sleep short period of time so we can wake up and check other things (e.g. check abort signal triggered). In this impl, the sleep time is 2 secs (which I think may still be too long).

Reply:
Changed to use 1.0 sec for sleep time and added explanation.